src/main/java/org/infernalstudios/infernalexp/blocks/GlowdustBlock.java

```diff
@@ -36,12 +92,58 @@ public BlockState getStateForPlacement(BlockPlaceContext context) {
         if (blockstate.is(this)) {
             int i = blockstate.getValue(LAYERS);
             if (i < 7) {
-                return blockstate.setValue(LAYERS, Integer.valueOf(Math.min(8, i + 1)));
+                return blockstate.setValue(LAYERS, Math.min(8, i + 1));
             } else {
                 return IEBlocks.GLOWDUST_SAND.get().defaultBlockState();
             }
         } else {
             return super.getStateForPlacement(context);
         }
     }
+
+    @Override
+    public boolean useShapeForLightOcclusion(BlockState state) {
+        return true;
+    }
+
+    @Override
+    public float getShadeBrightness(BlockState state, BlockGetter level, BlockPos pos) {
+        return state.getValue(LAYERS) == 8 ? 0.2F : 1.0F;
+    }
+
+    @Override
+    public boolean canSurvive(BlockState state, LevelReader level, BlockPos pos) {
+        BlockState blockstate = level.getBlockState(pos.below());
+        if (blockstate.is(BlockTags.SNOW_LAYER_CANNOT_SURVIVE_ON)) {
+            return false;
+        } else if (blockstate.is(BlockTags.SNOW_LAYER_CAN_SURVIVE_ON)) {
+            return true;
```

Comment:
It might be worth making a new block tag since the snow layer tags might now account for Nether blocks. This would be a simple as

```json
{
  "values": [
    "#snow_layer_can_survive_on",
    // [Whatever else should go here]
  ]
}
```

Reply:
Since this is copied from SnowLayerBlock, I didn't pay much attention to it.
I'd like to discard `snow_layer_cannot_survive_on`, and keep `snow_layer_can_survive_on`, since, despite the name, it's just a tag, which contains non-full blocks, that a layer could be placed on.